### ISSUE TITLE
python superdst.unpack() fails (Trac #499)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#498]
If you try to do frame['I3SuperDST'].unpack(), it gives this lovely error message:

TypeError: No to_python (by-value) converter found for C++ type: boost::shared_ptr<I3Map<OMKey, std::vector<I3RecoPulse, std::allocator<I3RecoPulse> > > const>

Test Case:

Open a file from IC86-2011, get to a Q frame, and try unpacking the I3SuperDST object.

```text
/cvmfs/icecube.wisc.edu/py2-v1/RHEL_6_x86_64/metaprojects/offline-software/trunk/env-shell.sh ipython
************************************************************************
*                                                                      *
*                   W E L C O M E  to  I C E T R A Y                   *
*                                                                      *
*              Version offline-software.trunk     r117764              *
*                                                                      *
*                You are welcome to visit our Web site                 *
*                        http://icecube.umd.edu                        *
*                                                                      *
************************************************************************

Icetray environment has:
   I3_SRC       = /cvmfs/icecube.wisc.edu/py2-v1/metaprojects/offline-software/trunk
   I3_BUILD     = /cvmfs/icecube.wisc.edu/py2-v1/RHEL_6_x86_64/metaprojects/offline-software/trunk
   I3_PORTS     = /cvmfs/icecube.wisc.edu/py2-v1/RHEL_6_x86_64/i3ports
   Python       = Python 2.7.3
Python 2.7.3 (default, Feb 20 2014, 23:54:47) 
Type "copyright", "credits" or "license" for more information.

IPython 1.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from icecube import icetray,dataio,dataclasses

In [2]: file = dataio.I3File('/data/exp/IceCube/2012/filtered/level2/0303/Level2_IC86.2011_data_Run00119744_Part00000000.i3.bz2')

In [3]: frame = file.pop_frame()

In [4]: frame = file.pop_frame()

In [5]: frame['I3SuperDST'].unpack()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-ffc3e2e09895> in <module>()
----> 1 frame['I3SuperDST'].unpack()

TypeError: No to_python (by-value) converter found for C++ type: boost::shared_ptr<I3Map<OMKey, std::vector<I3RecoPulse, std::allocator<I3RecoPulse> > > const>
```

Migrated from https://code.icecube.wisc.edu/ticket/499
```json
{
    "status": "closed", 
    "changetime": "2015-02-11T18:40:21", 
    "description": "If you try to do frame['I3SuperDST'].unpack(), it gives this lovely error message:\n\nTypeError: No to_python (by-value) converter found for C++ type: boost::shared_ptr<I3Map<OMKey, std::vector<I3RecoPulse, std::allocator<I3RecoPulse> > > const>\n\n\nTest Case:\n\nOpen a file from IC86-2011, get to a Q frame, and try unpacking the I3SuperDST object.\n\n{{{\n/cvmfs/icecube.wisc.edu/py2-v1/RHEL_6_x86_64/metaprojects/offline-software/trunk/env-shell.sh ipython\n************************************************************************\n*                                                                      *\n*                   W E L C O M E  to  I C E T R A Y                   *\n*                                                                      *\n*              Version offline-software.trunk     r117764              *\n*                                                                      *\n*                You are welcome to visit our Web site                 *\n*                        http://icecube.umd.edu                        *\n*                                                                      *\n************************************************************************\n\nIcetray environment has:\n   I3_SRC       = /cvmfs/icecube.wisc.edu/py2-v1/metaprojects/offline-software/trunk\n   I3_BUILD     = /cvmfs/icecube.wisc.edu/py2-v1/RHEL_6_x86_64/metaprojects/offline-software/trunk\n   I3_PORTS     = /cvmfs/icecube.wisc.edu/py2-v1/RHEL_6_x86_64/i3ports\n   Python       = Python 2.7.3\nPython 2.7.3 (default, Feb 20 2014, 23:54:47) \nType \"copyright\", \"credits\" or \"license\" for more information.\n\nIPython 1.1.0 -- An enhanced Interactive Python.\n?         -> Introduction and overview of IPython's features.\n%quickref -> Quick reference.\nhelp      -> Python's own help system.\nobject?   -> Details about 'object', use 'object??' for extra details.\n\nIn [1]: from icecube import icetray,dataio,dataclasses\n\nIn [2]: file = dataio.I3File('/data/exp/IceCube/2012/filtered/level2/0303/Level2_IC86.2011_data_Run00119744_Part00000000.i3.bz2')\n\nIn [3]: frame = file.pop_frame()\n\nIn [4]: frame = file.pop_frame()\n\nIn [5]: frame['I3SuperDST'].unpack()\n---------------------------------------------------------------------------\nTypeError                                 Traceback (most recent call last)\n<ipython-input-5-ffc3e2e09895> in <module>()\n----> 1 frame['I3SuperDST'].unpack()\n\nTypeError: No to_python (by-value) converter found for C++ type: boost::shared_ptr<I3Map<OMKey, std::vector<I3RecoPulse, std::allocator<I3RecoPulse> > > const>\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1423680021388075", 
    "component": "dataclasses", 
    "summary": "python superdst.unpack() fails", 
    "priority": "major", 
    "keywords": "", 
    "time": "2014-04-09T18:54:46", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20140917T14:58:40: 
* olivas changed milestone from "IceSim 4.2 "Spring Cleaning"" to ""
* olivas commented:

> Milestone IceSim 4.2 "Spring Cleaning" deleted

**Comment by icecube-trac** [comment]
Trac update at 20150211T18:40:21: 
* jvansanten commented:

> In changeset 128945:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="128945"
> Fix #498.
> ```


* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "new" to "closed"
